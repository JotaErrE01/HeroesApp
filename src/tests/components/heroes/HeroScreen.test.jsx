import { render, screen } from "@testing-library/react"
import { MemoryRouter, Route } from "react-router";
import HeroScreen from "../../../components/heroes/HeroScreen";
import userEvent from "@testing-library/user-event";

describe('Pruebas en <HeroScreen />', () => {

    const history = {
        length: 3,
        goBack: jest.fn(),
        push: jest.fn()
    }

    const param = 'dc-batman';
    
    test('Debe de mostrar el componente redirect si no hay argumentos en el url', () => {

        const { container } = render(
            <MemoryRouter initialEntries={['/hero']}>
                <HeroScreen history={history}/>
            </MemoryRouter>
        );

        expect( container.firstChild ).toBeNull();

    })

    test('Debe de mostrar un hero si el parametro existe', () => {

        const { container } = render(
            <MemoryRouter initialEntries={[`/hero/${param}`]} >
                <Route 
                    path='/hero/:heroId' 
                    component={HeroScreen} 
                />
            </MemoryRouter>
        );

        expect( container.firstChild ).toMatchSnapshot();

    })

    test('Debe de regresar a la pantalla anterior con push', () => {

        history.length = 2
      
        render(
            <MemoryRouter initialEntries={[`/hero/${param}`]}>
                <Route 
                    path='/hero/:heroId' 
                    component={ props => 
                        <HeroScreen 
                            {...props} 
                            history={history}
                        />
                    } 
                />
            </MemoryRouter>
        );

        userEvent.click(screen.getByText('Return'));

        expect( history.push ).toHaveBeenCalledWith('/');
        expect( history.goBack ).not.toHaveBeenCalled();

    })

    test('Debe de regresar a la pantalla anterior con goBak', () => {

        history.length = 3;
        
        render(
            <MemoryRouter initialEntries={[`/hero/${param}`]}>
                <Route  
                    path='/hero/:heroId' 
                    component={
                        props => <HeroScreen 
                            {...props}
                            history={history}
                        />
                    }
                />
            </MemoryRouter>
        );

        userEvent.click(screen.getByText('Return'));

        expect( history.push ).not.toHaveBeenCalled();
        expect( history.goBack ).toHaveBeenCalled();
    })
        
    
})
