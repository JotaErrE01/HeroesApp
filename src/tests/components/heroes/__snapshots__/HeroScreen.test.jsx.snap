// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroScreen /> Debe de mostrar un hero si el parametro existe 1`] = `
<div
  class="row mt-5"
>
  <div
    class="col-4"
  >
    <img
      alt="Batman"
      class="img-thumbnail animate__animated animate__fadeInLeft"
      src="/assets/heroes/dc-batman.jpg"
    />
  </div>
  <div
    class="col-8"
  >
    <h3>
      Batman
    </h3>
    <ul
      class="list-group list-group-flush"
    >
      <li
        class="list-group-item"
      >
         
        <b>
          Alter Ego: 
        </b>
         
        Bruce Wayne
         
      </li>
      <li
        class="list-group-item"
      >
         
        <b>
          Publisher 
        </b>
         
        DC Comics
         
      </li>
      <li
        class="list-group-item"
      >
         
        <b>
          First Apperance: 
        </b>
         
        Detective Comics #27
         
      </li>
    </ul>
    <h5>
       Characters 
    </h5>
    <p>
      Bruce Wayne
    </p>
    <button
      class="btn btn-outline-info"
    >
      Return
    </button>
  </div>
</div>
`;
